### DISCUSSION TITLE
Hooks  in React?

### DISCUSSION THREAD

**Discussion by mehulrajdev** [community/community#66387] [Apps, API and Webhooks]
### Select Topic Area

Question

### Body

How Many Hooks are Present in React?

**Comment by mehulrajdev** [comment]
React version 18 provides 15 hooks for developers. With 15 hook, you achieve similar functionality to a class-based component.

1 useCallback
2 useContext
3 useDebugValue
4 useDeferredValue
5 useEffect
6 useId
7 useImperativeHandle
8 useInsertionEffect
9 useLayoutEffect
10 useMemo
11 useReducer
12 useRef
13 useState
14 useSyncExternalStore
15 useTransition

**Comment by Akash1134** [comment]
GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this post. Please note any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
